lib/WWW/Spotify.pm

```diff
@@ -159,6 +159,34 @@ has ua => (
     },
 );
 
+has 'response_status' => (
+    is      => 'rw',
+    isa     => Str,
+    default => q{}
+);
+
+has 'response_content_type' => (
+    is      => 'rw',
+    isa     => Str,
+    default => q{}
+);
+
+has 'request_custom_handler' => (
+    is      => 'rw',
+    default => ''
+);
+
+has 'request_custom_handler_result' => (
+    is      => 'rw',
```

Comment:
You could make this `ro` and then use a `setter` method to write to it. That would allow you to update the attribute, but not the end user.

```perl
    writer => '_set_request_handler_result`
```

Reply:
good catch